stl/inc/spanstream

```diff
@@ -0,0 +1,396 @@
+#include <xutility>
+
+#pragma pack(push, _CRT_PACKING)
+#pragma warning(push, _STL_WARNING_LEVEL)
+#pragma warning(disable : _STL_DISABLED_WARNINGS)
+_STL_DISABLE_CLANG_WARNINGS
+#pragma push_macro("new")
+#undef new
+
+_STD_BEGIN
+
+template <class _Elem, class _Traits>
+class basic_spanbuf : public basic_streambuf<_Elem, _Traits> {
+public:
+    using _Mysb       = basic_streambuf<_Elem, _Traits>;
+    using char_type   = _Elem;
+    using int_type    = typename _Traits::int_type;
+    using pos_type    = typename _Traits::pos_type;
+    using off_type    = typename _Traits::off_type;
+    using traits_type = _Traits;
+
+    // [spanbuf.ctor], constructors
+    basic_spanbuf() = default;
+
+    explicit basic_spanbuf(ios_base::openmode _Which) : _Mysb(), _Mode(_Which) {}
+
+    explicit basic_spanbuf(_STD span<_Elem> _Span, ios_base::openmode _Which = ios_base::in | ios_base::out)
+        : _Mysb(), _Mode(_Which), _Buf(_Span) {
+        if (_Mode & ios_base::out) {
+            if (_Mode & ios_base::ate) {
+                _Mysb::setp(_Buf.data(), _Buf.data() + _Buf.size(), _Buf.data() + _Buf.size());
+            } else {
+                _Mysb::setp(_Buf.data(), _Buf.data(), _Buf.data() + _Buf.size());
+            }
+        }
+
+        if (_Mode & ios_base::in) {
+            _Mysb::setg(_Buf.data(), _Buf.data(), _Buf.data() + _Buf.size());
+        }
+    }
+
+    basic_spanbuf(const basic_spanbuf&) = delete;
+    basic_spanbuf(basic_spanbuf&& _Right)
+        : _Mysb(_STD move(_Right)), _Mode(_STD move(_Right._Mode)), _Buf(_STD move(_Right._Buf)) {
+        _Right.setp(nullptr, nullptr, nullptr);
+        _Right.setg(nullptr, nullptr, nullptr);
+    }
+
+    // [spanbuf.assign], assignment and swap
+    basic_spanbuf& operator=(const basic_spanbuf&) = delete;
+
+    basic_spanbuf& operator=(basic_spanbuf&& _Right) {
+        basic_spanbuf tmp{move(_Right)};
+        this->swap(tmp);
+        return *this;
+    }
+
+    void swap(basic_spanbuf& _Right) {
+        _Mysb::swap(_Right);
+        _STD swap(_Mode, _Right._Mode);
+        _STD swap(_Buf, _Right._Buf);
+    }
+
+    // [spanbuf.members], member functions
+    _STD span<_Elem> span() const noexcept {
+        if (_Mode & ios_base::out) {
+            return _STD span<_Elem>(_Mysb::pbase(), _Mysb::pptr());
+        }
+
+        return _Buf;
+    }
+
+    void span(_STD span<_Elem> _Span) noexcept {
+        _Buf = _Span;
+        if (_Mode & ios_base::out) {
+            if (_Mode & ios_base::ate) {
+                _Mysb::setp(_Buf.data(), _Buf.data() + _Buf.size(), _Buf.data() + _Buf.size());
+            } else {
+                _Mysb::setp(_Buf.data(), _Buf.data(), _Buf.data() + _Buf.size());
+            }
+        }
+
+        if (_Mode & ios_base::in) {
+            _Mysb::setg(_Buf.data(), _Buf.data(), _Buf.data() + _Buf.size());
+        }
+    }
+
+protected:
+    // [spanbuf.virtuals], overridden virtual functions
+    pos_type __CLR_OR_THIS_CALL seekoff(

```

Comment:
The occurrences of `__CLR_OR_THIS_CALL` in this file can be dropped - `/clr` is incompatible with C++20 (and C++23) so we can assume we're always compiling native code where this expands to nothing. `<syncstream>` overrode `overflow()` without this explicit calling convention (unlike `<fstream>` and `<strstream>` where it is definitely necessary), so omitting it here would be consistent with that new practice.

(The explicit calling convention is necessary for older code, to ensure that the virtual member function exactly matches the base class.)

Occurs below.